[Autograd Testing] Few refactors to test_autograd.py

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#35443 [Autograd Testing] Few refactors to test_autograd.py**
finishes.

Addressing Wanchao's comments from
https://github.com/pytorch/pytorch/pull/35268.

Differential Revision: [D20662292](https://our.internmc.facebook.com/intern/diff/D20662292/)

pritamdamania committed:
[Autograd Testing] Few refactors to test_autograd.py

Addressing Wanchao's comments from
https://github.com/pytorch/pytorch/pull/35268.

Differential Revision: [D20662292](https://our.internmc.facebook.com/intern/diff/D20662292/)

[ghstack-poisoned]

wanchaol reviewed (approved):
looks good, probably need a rebase?

pritamdamania committed:
Update on "[Autograd Testing] Few refactors to test_autograd.py"

finishes.

Addressing Wanchao's comments from
https://github.com/pytorch/pytorch/pull/35268.

Differential Revision: [D20662292](https://our.internmc.facebook.com/intern/diff/D20662292/)

[ghstack-poisoned]
